### PR TITLE
Show dependencies for casks

### PR DESCRIPTION
**Pull request by HaraldNordgren**
When running info on a cask show its dependencies.

This mirrors how it already works for formulas.

Fixes https://github.com/Homebrew/brew/issues/17600.

### CODE REVIEW

**Comment by HaraldNordgren** [comment]
@request-info, added description!

**Comment by apainintheneck** [comment]
This looks really good. I see that for formulae we also include a ✅ or ❌ depending on whether the dependency has already been installed. Do you think it's worth adding something like that for casks too?

```
==> Dependencies
Required: gettext ✔, pcre2 ✔
```

As is this PR is already a nice improvement. I just think it would be a nice-to-have thing. Not a blocker though.

File: Library/Homebrew/cask/info.rb
Comment (carlocab) [inline Library/Homebrew/cask/info.rb:86]:
```suggestion
      formula_deps = depends_on[:formula]&.map(&:to_s) || []
      cask_deps = depends_on[:cask]&.map { |dep| "#{dep} (cask)" } || []
```

**Review by carlocab — APPROVED** [review]
Thanks for your work here!

Some minor, optional, suggestions.

Also agree with the suggestion at https://github.com/Homebrew/brew/pull/17940#issuecomment-2267098019.

**Comment by HaraldNordgren** [comment]
I agree it would be great, but aybe it would be better to do that as a follow-up?

It seems the best way to achieve the goal would be to import this function from ` Library/Homebrew/cmd/info.rb`:

```
      def decorate_dependencies(dependencies)
        deps_status = dependencies.map do |dep|
          if dep.satisfied?([])
            pretty_installed(dep_display_s(dep))
          else
            pretty_uninstalled(dep_display_s(dep))
          end
        end
        deps_status.join(", ")
      end
```

But I don’t believe that that import is possible without causing a circular import. So that in turn would (probably) require the function to be moved to a shared location, where both places can import it. So it grows into much bigger PR.

Please correct me if I’m wrong.

**Review by apainintheneck — APPROVED** [review]

**Comment by apainintheneck** [comment]
Yeah, I think that can be handled as a follow-up.

**Comment by carlocab** [comment]
Yes, I'm fine with handling this as a follow-up.

Let's wait until the week begins to give other maintainers a chance to review this before we merge it.

File: Library/Homebrew/cask/info.rb
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/info.rb:86]:
```suggestion
      formula_deps = Array(depends_on[:formula]).map(&:to_s)
      cask_deps = Array(depends_on[:cask]).map { |dep| "#{dep} (cask)" }
```

**Review by MikeMcQuaid — APPROVED** [review]
Looks good, thanks @HaraldNordgren! I'm going to create a new tag later today so will hold off merging until after that.

Agreed that e.g. ✅ would be all of: nice to add, worth pulling into a shared location and worth doing in a follow-up PR.

**Comment by wickles** [comment]
Thanks for implementing this. Just want to note some missing features that would be nice to add later:

- `brew deps` implementation
- java dependencies

**Comment by HaraldNordgren** [comment]
@MikeMcQuaid I looked into how to possibly show the installation status for a cask in the same way as it is done for formulas. And the logic around `depends_on` differs a lot between casks and formulas.

I think I am in over my head a bit there.

**Comment by MikeMcQuaid** [comment]
@HaraldNordgren If you can open a draft PR with whatever you've done so far: it'll be easiest to help there when we're talking about code. I believe in you and can help you more ❤️

**Comment by HaraldNordgren** [comment]
Here is a WIP draft for the follow-up: https://github.com/Homebrew/brew/pull/17982

**Pull request merged**
